### PR TITLE
API V1: Add auth0

### PR DESCRIPTION
**Pull request by nuzurie**

### CODE REVIEW

**Comment by car12lin12** [comment]
can you add .idea folder to the gitignore please?

**Comment by car12lin12** [comment]
what are the new environment variables?

**Comment by car12lin12** [comment]
what are the new environment variables?

File: app/server.go [hunk 1]
```diff
@@ -31,13 +31,21 @@ func Server(
...
-	mapSchoolURLs(mw, schoolHandler, router)
+
+	mapStudentURLsV0(mwV0, studentHandler, router)
+	mapSchoolURLsV0(mwV0, schoolHandler, router)
 	mapTagURLs(tagHandler, router)
-	mapReviewURLs(mw, reviewHandler, router)
+	mapReviewURLsV0(mwV0, reviewHandler, router)
+
+	mapStudentURLsV1(mwV1, parser, studentHandler, router)
+	mapSchoolURLsV1(mwV0, parser, schoolHandler, router)
```
Comment (car12lin12) [inline app/server.go]: why do school and review urls V1 have V0 auth?

Comment (nuzurie) [reply app/server.go]: oops, that's a mistake on my part. I'll fix it

File: app/middlwares/Auth0Middleware.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		w.WriteHeader(http.StatusUnauthorized)
+		w.Write([]byte(`{"message":"Failed to validate JWT."}`))
+	}
+
+	middleware := jwtmiddleware.New(
+		jwtValidator.ValidateToken,
+		jwtmiddleware.WithErrorHandler(errorHandler),
+	)
+
+	return adapter.Wrap(middleware.CheckJWT)
```
Comment (car12lin12) [inline app/middlwares/Auth0Middleware.go:74]: why do you use an adapter?

**Review by car12lin12 — APPROVED** [review]
Great work Zubair :muscle:

File: app/middlwares/Auth0Middleware.go (hunk 2 above)
Comment (nuzurie) [reply app/middlwares/Auth0Middleware.go:74]: to convert golang net/http middleware to gin middleware :)

**Pull request merged**